src/controllers/userController.js

```diff
@@ -0,0 +1,130 @@
+import moment from 'moment';
+import crypto from 'crypto';
+import Validation from '../validation';
+import { sendResetMail, sendSignupMail } from '../services/sendMail';
+import { errorResponse, successResponse } from '../utils/response';
+import Hash from '../utils/hash';
+import models from '../models';
+
+const { User, Login, Reset } = models;
+
+export default class UserController {
+  /**
+   * @description Generate link to reset a user password
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns Promise {UserController} A reset link for new password
+   * @memberof UserController
+   */
+  static async forgotPassword(req, res) {
+    try {
+      const { errors, isValid } = Validation.validateEmail(req.body);
```

Comment:
Why are you validating in the controller? Validations, especially payload validation should be in middlewares. Your controller should just handle the routes main logic.

Reply:
@conquext are you resolving this 👆 ?

Reply:
I can raise a new ticket to resolve this but i am not able to make big changes right now

Reply:
Validations moved to middlewares.